pkg/cryptoutils/publickey.go

```diff
@@ -78,3 +81,24 @@ func SKID(pub crypto.PublicKey) ([]byte, error) {
 	skid := sha1.Sum(spki.SubjectPublicKey.Bytes) // nolint:gosec
 	return skid[:], nil
 }
+
+// EqualKeys compares two public keys. Supports RSA, ECDSA and ED25519.
+func EqualKeys(first, second crypto.PublicKey) error {
+	switch pub := first.(type) {
+	case *rsa.PublicKey:
```

Comment:
Is there any way we could do this with interfaces rather than the switch?

Reply:
I don't believe so. `crypto.PublicKey` doesn't define `Equals`, only the key types do.

If you mean using the PublicKey interface defined by Sigstore, I don't think it makes sense to use that since we'd have to initialize a verifier for both public keys to be able to compare them.

Reply:
Ooof, yeah you're right. This is something generics might actually help with! I keep trying to come up with use cases :)

This looks right for now, thanks.

Reply:
Very much agree! I'm also looking forward to getting to use generics once we upgrade to 1.18.